### PR TITLE
fix(parser): do not double-parse modifiers of ts index signature in class

### PR DESCRIPTION
**Pull request by wan9chi**
Modifiers of ts index signatures in classes are parsed twice. The first one is ignored and the second one is always empty, causing the incorrect `readonly` and `span` in ast. https://oxc-project.github.io/oxc/playground/?code=3YCAAICvgICAgICAgICxG4jI43W9aqTWr3Wzy1UcIvgVm0uDVLP%2FHYReSN%2FJ0Mhppm0XIo%2F7DW7cd39qCwCA

### CODE REVIEW

File: crates/oxc_parser/src/lib.rs [hunk 1]
```diff
@@ -696,6 +696,23 @@ mod test {
         assert_eq!(ret.errors.first().unwrap().to_string(), "Source length exceeds 4 GiB limit");
     }
 
+    #[test]
```
Comment (Boshen) [inline crates/oxc_parser/src/lib.rs:699]: This file is for integration tests; parser tests should go into https://github.com/oxc-project/oxc/tree/main/tasks/coverage/misc/pass

**Comment by Boshen** [comment]
I'm curious what project you are working on with `oxc` 😄

File: crates/oxc_parser/src/lib.rs (hunk 1 above)
Comment (branchseer) [reply crates/oxc_parser/src/lib.rs:699]: I think this fix can only be verified by examining the ast. It can't be done in there, can it?

**Comment by branchseer** [comment]
> I'm curious what project you are working on with `oxc` 😄

Just experimenting something 😄. Don't wanna spoil it now because I'm not sure if it's even viable. But if it is, I will surely let you guys know.

File: tasks/coverage/snapshots/parser_typescript.snap [hunk 2]
```diff
@@ -15102,38 +15148,6 @@ Expect to Parse: tasks/coverage/typescript/tests/cases/conformance/salsa/private
  9 │ }
    ╰────
 
-  × TS(1071): 'static' modifier cannot appear on an index signature.
-    ╭─[typescript/tests/cases/conformance/classes/staticIndexSignature/staticIndexSignature4.ts:12:5]
- 11 │ interface IB {
- 12 │     static [s: string]: number;
```
Comment (DonIsaac) [inline tasks/coverage/snapshots/parser_typescript.snap:15108]: Looks like you forgot to check if `static` is being used within an interface

**Pull request closed without merging**